pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -35,32 +35,49 @@ import (
 
 	appsv1alpha1 "github.com/karmada-io/karmada/pkg/apis/apps/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
 	"github.com/karmada-io/karmada/pkg/util/names"
 )
 
 const (
 	// ControllerName is the controller name that will be used when reporting events.
 	ControllerName = "workload-rebalancer"
+
+	ttlAfterFinishedWorkerNum = 1
 )
 
 // RebalancerController is to handle a rebalance to workloads selected by WorkloadRebalancer object.
 type RebalancerController struct {
-	Client client.Client
+	Client             client.Client // used to operate WorkloadRebalancer resources from cache.
+	ControlPlaneClient client.Client // used to fetch arbitrary resources from api server.
+
+	ttlAfterFinishedWorker util.AsyncWorker
```

Comment:
Smells not right, can't we check and remove the object in the reconcile loop?

Reply:
I didn't understand, can you give an example?

Reply:
ready to continue to review~ @RainbowMango